File: Clase6/SoftProgSolution/SoftProgModel/src/main/java/pe/edu/pucp/softprog/ventas/model/LineaOrdenVenta.java
package pe.edu.pucp.softprog.ventas.model;

import pe.edu.pucp.softprog.almacen.model.Producto;

public class LineaOrdenVenta {
    private int idLineaOrdenVenta;
    private Producto producto;
    private int cantidadUnidades;
    private double subtotal;
    private boolean activo;

    public int getIdLineaOrdenVenta() {
        return idLineaOrdenVenta;
    }

    public void setIdLineaOrdenVenta(int idLineaOrdenVenta) {
        this.idLineaOrdenVenta = idLineaOrdenVenta;
    }

    public Producto getProducto() {
        return producto;
    }

    public void setProducto(Producto producto) {
        this.producto = producto;
    }

    public int getCantidadUnidades() {
        return cantidadUnidades;
    }

    public void setCantidadUnidades(int cantidadUnidades) {
        this.cantidadUnidades = cantidadUnidades;
    }

    public double getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(double subtotal) {
        this.subtotal = subtotal;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    public void calcularSubtotal(){
        subtotal = cantidadUnidades * producto.getPrecio();
    }
}

File: Clase12/ProyectosNET/SoftProgSolution/SoftProgModel/Ventas/LineaOrdenVenta.cs
using SoftProgModel.Almacen;
using System;
using System.Collections.Generic;
using System.Text;

namespace SoftProgModel.Ventas
{
    public class LineaOrdenVenta
    {
        private int _idLineaOrdenVenta;
        private Producto _producto;
        private int _cantidadUnidades;
        private double _subtotal;
        private bool _activa;

        public int IdLineaOrdenVenta { get => _idLineaOrdenVenta; set => _idLineaOrdenVenta = value; }
        public Producto Producto { get => _producto; set => _producto = value; }
        public int CantidadUnidades { get => _cantidadUnidades; set => _cantidadUnidades = value; }
        public double Subtotal { get => _subtotal; set => _subtotal = value; }
        public bool Activa { get => _activa; set => _activa = value; }
    }
}
